presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java

```diff
@@ -151,9 +154,23 @@ void recordAddInput(OperationTimer operationTimer, Page page)
         }
     }
 
+    public void recordRawInput(long sizeInBytes, long positions)
+    {
+        rawInputDataSize.update(sizeInBytes);
+        rawInputPositions.update(positions);
+    }
+
+    public void recordRawInput(long sizeInBytes, long positions, long readNanos)
+    {
+        rawInputDataSize.update(sizeInBytes);
+        rawInputPositions.update(positions);
+        addInputTiming.record(readNanos, 0);
```

Comment:
Now since `addInputTiming` is used in both, `recordRawInput` and `recordGeneratedInput` it is very easy to missuse. Someone may accidently call both methods with `readNanos` provided. I would suggest updating the `addInputTiming` only in one of the methods (preferable `recordRawInput`), or maybe even extracting it into a separate method `recordReadWallNanos`

Reply:
How about having `recordRawInput(long sizeInBytes, long positions, long readNanos)`, `recordRawInput(long sizeInBytes, long positions)`, and only one `recordGeneratedInput(long sizeInBytes)` (removing the `position`parameter and the `recordGeneratedInput(long sizeInBytes, long positions, long readNanos)` overload)

Reply:
I ended up with:
```
recordRawInput(long sizeInBytes)
recordRawInputWithTiming(long sizeInBytes, long readNanos)
recordProcessedInput(long sizeInBytes, long positions)
```
`readNanos` is relevant to raw input because for source operators it translates into `rawInputReadTime` in `DriverContext#getDriverStats`